lsp/ruby_lsp.lua

```diff
@@ -20,8 +20,12 @@ return {
   init_options = {
     formatter = 'auto',
   },
+  on_new_config = function(config, root_dir)
```

Comment:
`on_new_config` is nothing a thing in vim.lsp.config. did you actually test this? 

https://github.com/neovim/nvim-lspconfig/pull/4081 shows an alternative approach for ensuring that a `cmd` has the correct CWD for LSes that (wrongly) care about that.